### PR TITLE
chore(wren-ai-service): improve ai service

### PR DESCRIPTION
**Pull request by cyyeh**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new target `use-wren-ui-as-engine` to streamline application startup.
	- Added new fields `rephrased_question` and `intent_reasoning` in `AskResultResponse` for enhanced query context.
	- Added a new function `update_config` to update YAML configuration for pipeline entries.
	- Implemented two new classes, `SQLBreakdownGenPostProcessor` and `SQLGenPostProcessor`, for improved SQL generation processing.

- **Bug Fixes**
	- Improved error handling in the `ask` method to preserve additional context.

- **Refactor**
	- Removed caching decorators from several functions to alter their performance behavior.
	- Restructured SQL processing capabilities by moving classes to a new module for better organization.
	- Updated import paths for several components to reflect new module structure.
	- Enhanced control over file reloads in the application startup process.

- **Chores**
	- Updated model configurations to include a `seed` parameter for randomness control.
	- Changed the format of questions data loading from JSON to YAML in tests.

### CODE REVIEW

**Comment by cyyeh** [comment]
@coderabbitai configuration

**Comment by cyyeh** [comment]
@coderabbitai full review

**Comment by cyyeh** [comment]
@coderabbitai full review

File: deployment/kustomizations/base/cm.yaml [hunk 1]
```diff
@@ -59,15 +59,17 @@ data:
...
+    - model: gpt-4o-2024-08-06
       api_base: https://api.openai.com/v1
       api_key_name: LLM_OPENAI_API_KEY
       kwargs:
         temperature: 0
         n: 1
+        seed: 0
         max_tokens: 4096
         response_format:
           type: json_object
```
[the remark this answers is not in the document]
Comment (cyyeh) [reply deployment/kustomizations/base/cm.yaml:75]: it's the correct one, please refer here: https://platform.openai.com/docs/models#current-model-aliases

**Review by paopa — APPROVED** [review]
LGTM

**Pull request merged**